iree-prof-tools/model-explorer-extension/src/modelExplorer.ts

```diff
@@ -82,12 +106,86 @@ export class WebviewPanelForModelExplorer {
 
     this.addDisposeCallback(() => { clearTimeout(timeout); });
   }
+
+  // Called on messages from web explorer in the webview.
+  private async onMessage(message: any) {
+    console.debug(
+      'Got message from model-explorer: ' + JSON.stringify(message, null, 2)
+    );
+    if (message.cmd == 'model-explorer-node-selected') {
+      const node = this.nodeMap?.getNodeById(message.nodeId);
+      if (node) {
+        this.focusOnText(node.label);
+      } else {
+        console.error('Unknown node ID: ' + message.nodeId);
+      }
+    }
+  }
+
+  // Sets focus on text matched with the node label of the current node of mode
+  // explorer in the webview.
+  private async focusOnText(nodeLabel: string | undefined) {
+    if (!nodeLabel || !this.editor) {
+      return;
+    }
+
+    // Find all matches with node label.
+    const document = this.editor.document;
+    const selections: vscode.Selection[] = [];
+    for (const match of document.getText().matchAll(RegExp(nodeLabel, 'g'))) {
```

Comment:
This may have a performance problem, but you can make it better later.
This is where LSP may be helpful too. (Assuming LSP also bookkeeping the source location in the editor.) Otherwise, we need to come up with a solution that caches the locations of symbols at least.

Reply:
Right. I plan to add source line:char info in graph json. This is for the case when json doesn't have that. Added a TODO.